builder/proxmox/step_finalize_template_config.go

```diff
@@ -56,6 +56,29 @@ func (s *stepFinalizeTemplateConfig) Run(ctx context.Context, state multistep.St
 		changes["ide2"] = "none,media=cdrom"
 	}
 
+	if c.CloudInit {
+		vmParams, err := client.GetVmConfig(vmRef)
+		if err != nil {
+			err := fmt.Errorf("Error fetching template config: %s", err)
+			state.Put("error", err)
+			ui.Error(err.Error())
+			return multistep.ActionHalt
+		}
+
+		if c.CloudInitStoragePool == "" {
+			if vmParams["bootdisk"] != nil && vmParams[vmParams["bootdisk"].(string)] != nil {
+				bootDisk := vmParams[vmParams["bootdisk"].(string)].(string)
+				c.CloudInitStoragePool = strings.Split(bootDisk, ":")[0]
```

Comment:
I think making changes to the config object is a quite confusing. Could you init a local variable outside the `if c.CloudInitStoragePool == ""` with `c.CloudInitStoragePool` and then conditionally change it if empty? I believe that would be easier to follow

Reply:
Done and squashed.